pallet-chainlink-feed/src/lib.rs

```diff
@@ -0,0 +1,1357 @@
+	}
+}
+
+/// Trait for interacting with the feeds in the pallet.
+pub trait FeedOracle {
+	type FeedId: Parameter + BaseArithmetic;
+	type Feed: FeedInterface;
+
+	/// Return the interface for the given feed if it exists.
+	fn feed(id: Self::FeedId) -> Option<Self::Feed>;
+}
+
+/// Trait for interacting with a particular feed.
+pub trait FeedInterface {
+	type AccountId: Parameter;
+	type BlockNumber: Parameter + BaseArithmetic;
+	type FeedId: Parameter + BaseArithmetic;
+	type Value: Parameter + BaseArithmetic;
+
+	/// Returns the id of the first round that contains non-default data.
+	fn first_valid_round(&self) -> Option<RoundId>;
+
+	/// Returns the id of the latest oracle round.
+	fn latest_round(&self) -> RoundId;
+
+	/// Returns the data for a given round.
+	/// Will return `None` if there is no data for the given round.
+	fn data_at(&self, round: RoundId) -> Option<RoundData<Self::BlockNumber, Self::Value>>;
+
+	/// Returns the latest data for the feed.
+	/// Will always return data but may contain default data if there
+	/// has not been a valid round, yet.
+	/// Check `first_valid_round` to determine whether there is
+	/// useful data yet.
+	fn latest_data(&self) -> RoundData<Self::BlockNumber, Self::Value>;
+
+	/// Request that a new oracle round be started.
+	///
+	/// **Warning:** Fallible function that changes storage.
+	fn request_new_round(&mut self, requester: Self::AccountId) -> DispatchResult;
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as ChainlinkFeed {
+		/// The account controlling the funds for this pallet.
+		pub PalletAdmin get(fn pallet_admin) config(): T::AccountId;
+		// possible optimization: put together with admin?
+		/// The account to set as future pallet admin.
+		pub PendingPalletAdmin: Option<T::AccountId>;
+
+		/// Tracks the amount of debt accrued by the pallet towards the oracles.
+		pub Debt get(fn debt): BalanceOf<T>;
+
+		/// A running counter used internally to determine the next feed id.
+		pub FeedCounter get(fn feed_counter): T::FeedId;
+
+		/// Configuration for a feed.
+		pub Feeds get(fn feed_config): map hasher(twox_64_concat) T::FeedId => Option<FeedConfigOf<T>>;
+
+		/// Accounts allowed to create feeds.
+		pub FeedCreators: map hasher(blake2_128_concat) T::AccountId => Option<()>;
+
+		/// User-facing round data.
+		pub Rounds get(fn round): double_map hasher(twox_64_concat) T::FeedId, hasher(twox_64_concat) RoundId => Option<RoundOf<T>>;
+
+		/// Operator-facing round data.
+		pub Details get(fn round_details): double_map hasher(twox_64_concat) T::FeedId, hasher(twox_64_concat) RoundId => Option<RoundDetailsOf<T>>;
+
+		/// Global oracle meta data including admin and withdrawable funds.
+		pub Oracles get(fn oracle): map hasher(blake2_128_concat) T::AccountId => Option<OracleMetaOf<T>>;
+
+		/// Feed local oracle status data.
+		pub OracleStati get(fn oracle_status): double_map hasher(twox_64_concat) T::FeedId, hasher(blake2_128_concat) T::AccountId => Option<OracleStatusOf<T>>;
```

Comment:
`OracleStati` -> `OracleStatus`?

Reply:
The storage is normally plural (like `Oracles` above)

Reply:
of course open to better naming ideas

Reply:
Where does `Stati` comes from?  Afaik, `Status` is plural of itself and most used than `Statuses`.

Reply:
Latin xD (it's how we get the plural for Latin words in German as well)

Reply:
went for Statuses to make the distinction clearer